Crashlytics/Crashlytics/Models/Record/FIRCLSRecordBase.m

```diff
@@ -0,0 +1,38 @@
+/*
+ * Copyright 2020 Google
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#import "FIRCLSRecordBase.h"
+
+@implementation FIRCLSRecordBase
+
+- (instancetype)initWithDict:(NSDictionary *)dict {
+  return [super init];
+}
+
++ (NSString *)decodedHexStringWithValue:(NSString *)hexEncodedString {
+  NSMutableString *decodedString = [[NSMutableString alloc] init];
+  int index = 0;
+  while (index < hexEncodedString.length) {
+    NSString *hexChar = [hexEncodedString substringWithRange:NSMakeRange(index, 2)];
+    int value = 0;
+    sscanf([hexChar cStringUsingEncoding:NSASCIIStringEncoding], "%x", &value);
+    [decodedString appendFormat:@"%c", (char)value];
+    index += 2;
+  }
+  return decodedString;
```

Comment:
Just a thought, I 🤔 if 👻s will trip this up?

Reply:
We might want to track test cases in a bug or something

Reply:
Created and shared internal test case spreadsheet.